src/System.Private.ServiceModel/src/System/ServiceModel/Channels/MessageContent.cs

```diff
@@ -186,7 +186,7 @@ protected override Task<Stream> CreateContentReadStreamAsync()
 
         protected override Task SerializeToStreamAsync(Stream stream, TransportContext context)
         {
-            return _messageEncoder.WriteMessageAsync(_message, _stream);
+            return _messageEncoder.WriteMessageAsync(_message, stream);
```

Comment:
The old code seems to be a bug. I checked this with Matt and he confirmed this was a bug. I found this because I got a NullRef exception here. Those two variables are not pointing to the same object.